src/DSharp.Mscorlib/Collections/Generic/Dictionary.cs

```diff
@@ -8,44 +8,57 @@ namespace System.Collections.Generic
     [ScriptIgnoreNamespace]
     [ScriptImport]
     [ScriptName("Object")]
-    public sealed class Dictionary<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
+    public sealed partial class Dictionary<TKey, TValue>
+        : IDictionary<TKey, TValue>
+        , IDictionary
+        , IReadOnlyDictionary<TKey, TValue>
     {
         public Dictionary() { }
 
-        public Dictionary(params object[] nameValuePairs) { }
+        public extern ICollection<TKey> Keys { get; }
 
-        [Obsolete("This is only for use by the c# compiler, and cannot be used for generating script.", /* error */ true)]
-        public extern Dictionary(int count);
+        extern ICollection IDictionary.Keys { get; }
 
-        public extern int Count { get; }
+        extern IEnumerable<TKey> IReadOnlyDictionary<TKey, TValue>.Keys { get; }
 
-        public extern IReadonlyCollection<TKey> Keys { get; }
+        public extern ICollection<TValue> Values { get; }
+
+        extern ICollection IDictionary.Values { get; }
+
+        extern IEnumerable<TValue> IReadOnlyDictionary<TKey, TValue>.Values { get; }
 
         [ScriptField]
         public extern TValue this[TKey key] { get; set; }
 
-        [Obsolete("This is only for use by the c# compiler, and cannot be used for generating script.", /* error */ true)]
+        [ScriptField]
+        public extern object this[object key] { get; set; }
+
+        [DSharpScriptMemberName("addKeyValue")]
         public extern void Add(TKey key, TValue value);
 
+        [DSharpScriptMemberName("addKeyValue")]
+        public extern void Add(object key, object value);
+
+        [Obsolete("This is only for use by the c# compiler, and cannot be used for generating script.", error: true)]
```

Comment:
Wonder if we can make this internal and have the compiler look for the internal symbol instead. Be nicer than exposing a method we can't use in the stdlib

Reply:
this needs to be here because of the contract from `ICollection` interface. If I make it internal, the contract isn't fulfilled

Reply:
That sucks :(